scheduler-server/build.gradle

```diff
@@ -9,8 +10,48 @@ java {
   }
 }
 
+node {
+  nodeProjectDir = file("${project.projectDir.toPath().resolve("scheduling-dsl-compiler")}")
+  version = '16.4.0'
+  npmVersion = '8.5.0'
+  download = true
+}
+
+task assembleSchedulingDSLCompiler(type: NpmTask) {
+  dependsOn processResources
+  dependsOn npmInstall
+  args = ['run', 'build']
+  inputs.files('package.json', 'package-lock.json', 'tsconfig.json')
+  inputs.dir('src')
+}
+
+task generateASTJsonSchema(type: NpmTask) {
+  dependsOn processResources
+  dependsOn npmInstall
+  args = ['run', 'generate-ast-schema']
+  inputs.files('package.json', 'package-lock.json', 'tsconfig.json')
+  inputs.dir('src')
+}
+
+assemble {
+  dependsOn assembleSchedulingDSLCompiler
+  dependsOn generateASTJsonSchema
+}
+
 test {
+  dependsOn(assembleSchedulingDSLCompiler)
   useJUnitPlatform()
+  def architectureSpecificDir = new ByteArrayOutputStream()
+  exec{
+    commandLine "ls", projectDir.toPath().resolve(".gradle/nodejs")
+    standardOutput = architectureSpecificDir;
+    ignoreExitValue true
+  }
```

Comment:
So `architectureSpecificDir` contains the output from `ls .gradle/nodejs`? Not sure I fully understand what's being done above and below here, could you briefly describe what's needed here? Thanks :)

Reply:
Good question - if node is not available on a system, the gradle plugin will install node locally at this location: ```
scheduler-server/.gradle/nodejs/node-v16.4.0-darwin-x64/bin/node
```

I want to add `scheduler-server/.gradle/nodejs/node-v16.4.0-darwin-x64/bin/` to the PATH, so that java code can just expect `node` to be on the PATH.

Rather than hard-coding `node-v16.4.0-darwin-x64`, I get it via `ls .gradle/nodejs/`.

Reply:
Got it. I think this should be possible without bailing out to the shell. I'll mess with this and try to comment with a block that should work.

Reply:
> if node is not available on a system, the gradle plugin will install node locally at this location

And if node is available on the system then `scheduler-server/.gradle/nodejs/node-v16.4.0-darwin-x64` should not exist, right? If so, I think line 51 would result in an exception.

Reply:
Maybe `.gradle/nodejs/<node>` is generated regardless of whether node is installed on the system or not.

Reply:
Yeah, I think I ran into this issue at some point and got around it eventually by removing my local node from the path temporarily for it to provoke the install

Reply:
@mattdailis maybe we can simplify things here by assuming that the user has node installed on their system. After-all, it is a req. for aerie development (https://github.com/NASA-AMMOS/aerie/blob/develop/docs/building.md should probably be updated to reflect this).

Reply:
> Yeah, I think I ran into this issue at some point and got around it eventually by removing my local node from the path temporarily for it to provoke the install

Hmm, that's not ideal - I was hoping it would "just work" if you already had node.



> @mattdailis maybe we can simplify things here by assuming that the user has node@16 installed on their system. After-all, it is a req. for aerie development (https://github.com/NASA-AMMOS/aerie/blob/develop/docs/building.md should probably be updated to reflect this).

I'd be okay with this - I'd just need your help making sure that when our tests run as a github action, node is available.

Reply:
Created https://github.com/NASA-AMMOS/aerie/pull/76 which I think should resolve/simplify this.